index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
But are you quoting that in regards to some inner Semigroup or to the `Maybe`-wrapped one?   Are you trying to define types such as

    Maybe (String);  
    // Maybe ('ab').concat (Maybe ('cd')); //=> Maybe ('abcd')
    Maybe (List)
    // Maybe ([1, 2]).concat (Maybe (['foo'])); //=> Maybe ([1, 2, 'foo'])
    Maybe (Add)
    // Maybe(Add (3)).concat(Maybe (Add (4))):  //=> Maybe (Add (7))
    Maybe (Multiply)
    // Maybe (Multiply (3)).concat (Maybe (Multiply (4))): // Maybe (Multiply (12))

It seems that is what you would want if you are doing this internal `concat`.  But I are nothing in the implementation that attempts to enforce this sort of stronger typing.